chirho/robust/internals/linearize.py

```diff
@@ -138,6 +140,7 @@ def linearize(
         model, guide, num_samples=num_samples_inner, max_plate_nesting=max_plate_nesting
     )
     log_prob_params, func_log_prob = make_functional_call(log_prob)
+    func_log_prob = SeedMessenger(seed)(func_log_prob)
```

Comment:
This will fix the seed at the level of `linearize`, so that `linearize` and `influence_fn` return deterministic functions even though they are computing (unbiased) random quantities. Is that really what we want?

I thought what was necessary to address #407 was only that the seed to stay fixed for the duration of a single `cg_solver` call, which would mean changing the output `_fn` of `linearize` like so:
```python
def _fn(point: Point[T], *args: P.args, **kwargs: P.kwargs) -> ParamDict:
    ...
    with SeedMessenger(get_rng_seed()):
        return cg_solver(fvp, point_score)

return _fn
```

Reply:
How about using a helper function to pin the rng state across `fvp` calls in `cg_solver`?
```python
@contextlib.contextmanager
def reset_rng_state(rng_state):
    try:
        prev_rng_state = get_rng_state()
        set_rng_state(rng_state)
    finally:
        set_rng_state(rng_state)

...

def linearize(...):
    ...
    def _fn(point: Point[T], *args, **kwargs):
        ...
        pinned_fvp = reset_rng_state(get_rng_state())
        return cg_solver(pinned_fvp, point_score)

    return _fn
```

Reply:
Cool thanks let me give this a try!